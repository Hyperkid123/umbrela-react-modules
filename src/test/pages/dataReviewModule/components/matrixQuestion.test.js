import React from 'react'
import MatrixQuestion from '../../../../pages/dataReviewModule/components/matrixQuestion';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Matrix question component', () => {
  const initialProps = {}
  beforeEach(() => {
    initialProps.data = {
      dividedData: [{
        questionTitle: 'title',
        respondents: 100,
        answerCount: 33
      }]
    };
    initialProps.showLegend = true;
    initialProps.chartType = 'PIE_CHART';
    initialProps.question = {}
    initialProps.isOptionImage = false;
  });

  it('should render correctly', () => {
    const tree = shallow(
      <MatrixQuestion {...initialProps}/>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })

  it('should render correctly with images as subquestion', () => {
    const tree = shallow(
      <MatrixQuestion {...initialProps} isOptionImage={true}/>
    )
    expect(toJson(tree)).toMatchSnapshot()
  })
});
