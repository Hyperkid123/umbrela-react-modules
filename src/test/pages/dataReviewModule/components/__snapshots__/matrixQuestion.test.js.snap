// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matrix question component should render correctly 1`] = `
<styled.div
  column={true}
  grow={true}
>
  <CollapsibleSection
    key="0"
    sectionTitle="title"
  >
    <styled.div>
      <CloseQuestions
        chartType="PIE_CHART"
        data={
          Object {
            "answerCount": 33,
            "questionTitle": "title",
            "respondents": 100,
          }
        }
        noAnimation={true}
        showLegend={true}
        url={false}
      />
      <Connect(LegendTable)
        data={
          Object {
            "answerCount": 33,
            "barData": Object {
              "answerCount": 33,
              "questionTitle": "title",
              "respondents": 100,
            },
            "respondents": 100,
          }
        }
      />
    </styled.div>
  </CollapsibleSection>
</styled.div>
`;

exports[`Matrix question component should render correctly with images as subquestion 1`] = `
<styled.div
  column={true}
  grow={true}
>
  <CollapsibleSection
    key="0"
    sectionTitle="title"
  >
    <styled.div>
      <CloseQuestions
        chartType="PIE_CHART"
        data={
          Object {
            "answerCount": 33,
            "questionTitle": "title",
            "respondents": 100,
          }
        }
        noAnimation={true}
        showLegend={true}
        url="title"
      />
      <Connect(LegendTable)
        data={
          Object {
            "answerCount": 33,
            "barData": Object {
              "answerCount": 33,
              "questionTitle": "title",
              "respondents": 100,
            },
            "respondents": 100,
          }
        }
      />
    </styled.div>
  </CollapsibleSection>
</styled.div>
`;
